src/screens/title.rs

```diff
@@ -10,32 +10,27 @@ pub(super) fn plugin(app: &mut App) {
 }
 
 fn show_title_screen(mut commands: Commands) {
-    let enter_playing = commands.register_one_shot_system(enter_playing);
-    let enter_credits = commands.register_one_shot_system(enter_credits);
-    #[cfg(not(target_family = "wasm"))]
-    let exit_app = commands.register_one_shot_system(exit_app);
-
     commands
         .ui_root()
         .insert(StateScoped(Screen::Title))
         .with_children(|children| {
-            children.button("Play", enter_playing);
-            children.button("Credits", enter_credits);
+            children.button("Play").observe(enter_playing);
+            children.button("Credits").observe(enter_credits);
 
             #[cfg(not(target_family = "wasm"))]
-            children.button("Exit", exit_app);
+            children.button("Exit").observe(exit_app);
         });
 }
 
-fn enter_playing(mut next_screen: ResMut<NextState<Screen>>) {
+fn enter_playing(_trigger: Trigger<OnPress>, mut next_screen: ResMut<NextState<Screen>>) {
     next_screen.set(Screen::Playing);
 }
 
-fn enter_credits(mut next_screen: ResMut<NextState<Screen>>) {
+fn enter_credits(_trigger: Trigger<OnPress>, mut next_screen: ResMut<NextState<Screen>>) {
     next_screen.set(Screen::Credits);
 }
 
 #[cfg(not(target_family = "wasm"))]
-fn exit_app(mut app_exit: EventWriter<AppExit>) {
+fn exit_app(_trigger: Trigger<OnPress>, mut app_exit: EventWriter<AppExit>) {
     app_exit.send(AppExit::Success);
 }
```

Comment:
Could we use closures for this?

Reply:
A button setting a state seems like a pretty generic use case

Reply:
I would not do that in this case, as I want to teach the user how to do stuff on button presses in general, and a closure wrapper would obscure that somewhat.

Reply:
Can we replace just one of those, to show the generic change state or at least mention it in a comment how it scales?
Users might try to re-use the same functions between different themes, which is much worse than a single generic system
And you can have multiple observers on a button press, so adding more logic doesn't invalidate this case

Reply:
I'll accept "it doesn't need to scale" since most games have very simple navigation, just seeing if you would consider it

Reply:
pyri_state provides `Screen::Credits.enter()` for this btw. seems like something bevy_state should just provide instead.

Reply:
basically the same thing 🤔 
I feel like functions returning closures is already a pattern so I'd personally stick with that

As for `bevy_state` relying on observers, good idea, I'd just wait for observers to stabilize a bit more

Reply:
[`4529b0f` (#249)](https://github.com/TheBevyFlock/bevy_quickstart/pull/249/commits/4529b0fb7a9cd9fbd0a372f85952560ebd022d37)